### PR TITLE
Fix playlist deletion and a few warning fixes

### PR DESCRIPTION
**Pull request by dkanada**
Playlists are stored in directories, but there was a check for `File.Exists()` which doesn't return true for folders. @Bond-009 I tracked the issue back to a [commit](https://github.com/jellyfin/jellyfin/commit/ffe79c89829fb232e8ccb4ae4caf4b732ce51600) you made in January. I'm guessing this is the source of most of our deletion issues so we might need to go through and fix all the other changes from that commit and any similar changes.

### CODE REVIEW

File: Emby.Server.Implementations/Playlists/PlaylistManager.cs [hunk 1]
```diff
@@ -56,10 +56,8 @@ public async Task<PlaylistCreationResult> CreatePlaylist(PlaylistCreationRequest
         {
             var name = options.Name;
 
-            var folderName = _fileSystem.GetValidFilename(name) + " [playlist]";
-
+            var folderName = _fileSystem.GetValidFilename(name);
```
Comment (cvium) [inline Emby.Server.Implementations/Playlists/PlaylistManager.cs]: Why was this removed?

Comment (dkanada) [reply Emby.Server.Implementations/Playlists/PlaylistManager.cs]: This was just a style change. All internal playlists are stored in a folder called `playlists` so the additional string is redundant.

File: Emby.Server.Implementations/Library/LibraryManager.cs [hunk 2]
```diff
@@ -394,7 +394,7 @@ public void DeleteItem(BaseItem item, DeleteOptions options, BaseItem parent, bo
 
                 foreach (var fileSystemInfo in item.GetDeletePaths().ToList())
                 {
-                    if (File.Exists(fileSystemInfo.FullName))
+                    if (File.Exists(fileSystemInfo.FullName) || Directory.Exists(fileSystemInfo.FullName))
```
Comment (Bond-009) [inline Emby.Server.Implementations/Library/LibraryManager.cs]:
```suggestion
                    if (fileSystemInfo.IsDirectory ? Directory.Exists(fileSystemInfo.FullName) : File.Exists(fileSystemInfo.FullName))
```

File: Emby.Server.Implementations/Library/LibraryManager.cs [hunk 3]
```diff
@@ -394,7 +394,7 @@ public void DeleteItem(BaseItem item, DeleteOptions options, BaseItem parent, bo
 
                 foreach (var fileSystemInfo in item.GetDeletePaths().ToList())
```
Comment (Bond-009) [inline Emby.Server.Implementations/Library/LibraryManager.cs]:
```suggestion
                foreach (var fileSystemInfo in item.GetDeletePaths())
```

Comment (Bond-009) [reply Emby.Server.Implementations/Library/LibraryManager.cs]: Looks like this added an extra space?

File: Emby.Server.Implementations/Library/LibraryManager.cs [hunk 4]
```diff
@@ -392,9 +392,9 @@ public void DeleteItem(BaseItem item, DeleteOptions options, BaseItem parent, bo
                 // Add this flag to GetDeletePaths if required in the future
                 var isRequiredForDelete = true;
 
-                foreach (var fileSystemInfo in item.GetDeletePaths().ToList())
+                 foreach (var fileSystemInfo in item.GetDeletePaths())
```
Comment (Bond-009) [inline Emby.Server.Implementations/Library/LibraryManager.cs]:
```suggestion
                foreach (var fileSystemInfo in item.GetDeletePaths())
```
No idea how this happend

File: Emby.Server.Implementations/Library/LibraryManager.cs [hunk 5]
```diff
@@ -392,9 +392,9 @@ public void DeleteItem(BaseItem item, DeleteOptions options, BaseItem parent, bo
                 // Add this flag to GetDeletePaths if required in the future
                 var isRequiredForDelete = true;
 
-                foreach (var fileSystemInfo in item.GetDeletePaths().ToList())
+                foreach (var fileSystemInfo in item.GetDeletePaths())
                 {
-                    if (File.Exists(fileSystemInfo.FullName))
+                    if (fileSystemInfo.IsDirectory ? Directory.Exists(fileSystemInfo.FullName) : File.Exists(fileSystemInfo.FullName))
```
Comment (JustAMan) [inline Emby.Server.Implementations/Library/LibraryManager.cs]: Can we wrap it or change into several statements? I am not a fan of such complex ternaries...

Comment (dkanada) [reply Emby.Server.Implementations/Library/LibraryManager.cs]: I don't think this ternary is particularly confusing, and I think further nesting wouldn't be a better solution, but if you still think it should be changed I suppose I can. Do you have a good suggestion?
